### PR TITLE
[Work in Progress] Clarify error for opaque type in async functions

### PR DESCRIPTION
**Pull request by nellshamrell**
Still a work in progress.

Fix for #80658

### CODE REVIEW

File: compiler/rustc_infer/src/infer/error_reporting/mod.rs [hunk 1]
```diff
@@ -1490,7 +1490,7 @@ impl<'a, 'tcx> InferCtxt<'a, 'tcx> {
                             format!(
                                 "{}{}{} {}{}",
                                 if sp.is_desugaring(DesugaringKind::Async) {
-                                    "the `Output` of this `async fn`'s "
+                                    "checked the return type of this `async fn`, "
```
Comment (jyn514) [inline compiler/rustc_infer/src/infer/error_reporting/mod.rs:1493]: This doesn't look any more clear to me. Maybe add ``.note("while checking the return type of this `async fn`")`` instead?

Comment (nellshamrell) [reply compiler/rustc_infer/src/infer/error_reporting/mod.rs:1493]: That does make sense! I'll get that incorporated in.

**Comment by nellshamrell** [comment]
Closing in favor of #82165

File: compiler/rustc_infer/src/infer/error_reporting/mod.rs (hunk 1 above)
Comment (nellshamrell) [reply compiler/rustc_infer/src/infer/error_reporting/mod.rs:1493]: Done! Moved the changes to #82165

**Pull request closed without merging**